hydra-node/src/Hydra/Chain/Backend.hs

```diff
@@ -87,31 +91,55 @@ buildTransactionWithPParams' ::
 buildTransactionWithPParams' pparams systemStart eraHistory stakePools changeAddress utxoToSpend collateral outs = do
```

Comment:
i'm a little worried about having touched this function as it is used by `buildScriptPublishingTxs`.

are u sure these changes haven't affected the `publishHydraScripts` ?